engines/griffon/draw.cpp

```diff
@@ -0,0 +1,1230 @@
+						rcDest.top = npy;
+						rcDest.setWidth(24);
+						rcDest.setHeight(24);
+
+						_animsAttack[sprite]->blit(*_videoBuffer, rcDest.left, rcDest.top, Graphics::FLIP_NONE, &rcSrc);
+					}
+
+				}
+
+				// onewing
+				if (_npcInfo[i].spriteset == kMonsterOneWing) {
+					for (int f = 0; f < 7; f++) {
+						int s = _npcInfo[i].bodysection[f].sprite;
+						rcSrc.left = _animSet2[s].x;
+						rcSrc.top = _animSet2[s].y;
+						rcSrc.setWidth(_animSet2[s].w);
+						rcSrc.setHeight(_animSet2[s].h);
+
+						rcDest.left = _npcInfo[i].bodysection[f].x - _animSet2[s].xofs;
+						rcDest.top = _npcInfo[i].bodysection[f].y - _animSet2[s].yofs + 2;
+
+						_anims[2]->blit(*_videoBuffer, rcDest.left, rcDest.top, Graphics::FLIP_NONE, &rcSrc);
+					}
+
+				}
+
+				// twowing
+				if (_npcInfo[i].spriteset == kMonsterTwoWing) {
+					for (int f = 0; f < 7; f++) {
+						int yp = 0;
+
+						if (f == 0 && (_curMap == 53 || _curMap == 57 || _curMap == 61 || _curMap == 65 || _curMap == 56 || _curMap > 66) && _scriptFlag[kScriptLever][0] > 0)
+							yp = 16;
+						int s = _npcInfo[i].bodysection[f].sprite;
+						rcSrc.left = _animSet9[s].x;
+						rcSrc.top = _animSet9[s].y + yp;
+						rcSrc.setWidth(_animSet9[s].w);
+						rcSrc.setHeight(_animSet9[s].h);
+
+						rcDest.left = _npcInfo[i].bodysection[f].x - _animSet9[s].xofs;
+						rcDest.top = _npcInfo[i].bodysection[f].y - _animSet9[s].yofs + 2;
+
+						_anims[9]->blit(*_videoBuffer, rcDest.left, rcDest.top, Graphics::FLIP_NONE, &rcSrc);
+					}
+
+				}
+
+
+				//  boss 1
+				if (_npcInfo[i].spriteset == kMonsterBoss1) {
+					if (!_npcInfo[i].attacking) {
+						int cframe = _npcInfo[i].cframe;
+						rcSrc.left = (int)(cframe / 4) * 24;
+					} else {
+						rcSrc.left = 4 * 24;
+					}
+
+					rcSrc.top = 0;
+					rcSrc.setWidth(24);
+					rcSrc.setHeight(48);
+
+					rcDest.left = npx - 2;
+					rcDest.top = npy - 24;
+
+					_anims[3]->blit(*_videoBuffer, rcDest.left, rcDest.top, Graphics::FLIP_NONE, &rcSrc);
+
+				}
+
+				// black knight
+				if (_npcInfo[i].spriteset == kMonsterBlackKnight) {
+					if (!_npcInfo[i].attacking) {
+						int cframe = _npcInfo[i].cframe;
+						rcSrc.left = (int)(cframe / 4) * 24;
+					} else {
+						rcSrc.left = 4 * 24;
+					}
+					rcSrc.top = 0;
+					rcSrc.setWidth(24);
+					rcSrc.setHeight(48);
+
+					rcDest.left = npx - 2;
+					rcDest.top = npy - 24;
+
+					_anims[4]->blit(*_videoBuffer, rcDest.left, rcDest.top, Graphics::FLIP_NONE, &rcSrc);
+				}
+
+
+				// firehydra
+				if (_npcInfo[i].spriteset == kMonsterFireHydra) {
+					for (int ff = 0; ff <= 2; ff++) {
+						if (_npcInfo[i].hp > 10 * ff * 20) {
+							rcSrc.left = 16 * (int)(RND() * 2);
+							rcSrc.top = 80;
+							rcSrc.setWidth(16);
+							rcSrc.setHeight(16);
+
+							rcDest.left = _npcInfo[i].bodysection[10 * ff].x - 8;
+							rcDest.top = _npcInfo[i].bodysection[10 * ff].y - 8;
+
+							int x = 192 + ((int)(_itemyloc + ff * 5) % 3) * 64;
+							if (x > 255)
```

Comment:
If there are a lot of boundary checks like this in the engine, consider maybe using the MIN macro instead of separate if checks

Reply:
I am a bit torn on this one. Adding it will improve the code reability only slightly, but will require tons of work...

Reply:
Don't worry about it. After the further review I did of the remainder of the engine, there weren't really enough occurrences for me to be concerned about it.